tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -9465,7 +9465,7 @@ SpirvEmitter::processIntrinsicAsType(const CallExpr *callExpr) {
   switch (numArgs) {
   case 1: {
     // Handling Method 1, 2, and 3.
-    auto *argInstr = doExpr(arg0);
+    auto *argInstr = loadIfGLValue(arg0);
```

Comment:
Without this change, it directly uses `doExpr(arg0)` that is a pointer to argument. We have to make sure the argument is a r-value not a pointer.